### ISSUE TITLE
Spyder 5.2 Windows installer crashes during launch

### ISSUE THREAD

**Issue by wCTcTU4P** [spyder-ide/spyder#16896]
## Problem Description
Spyder 5.2. installed from Spyder_64bit_full.exe crashes during launch: it will load the GUI, a script already open (from previous windows .exe installations, which have never had this issue), and then after an infinitesimal moment with the "Issue Reporter" window open it closes. The console never loads, and the GUI is open for a very small amount of time. It does this every time I try to launch. 

I don't use Anaconda. I have a python 3.9 pipenv environment with PyQt5 = "==5.12.3" and spyder-kernels = "==2.1.0" and I've had no issues using the Spyder windows installer version of Spyder with that env for months (installing via pip did result in issues). Tried switching to "just for me" install vs. "everyone", and that didn't work. 

I tried uninstalling Spyder and reinstalling with the new .exe file, but it made no difference, nor did running as an administrator. I'm submitting this because I've installed a half dozen or more of the most recent v5+ https://www.spyder-ide.org/ windows .exe Spyder versions as they release, and this has never occurred. Nothing novel was done on my end.

**Comment by ccordoba12** [comment]
Hey @wCTcTU4P, thanks for reporting. We'll take a look at this as soon as we can. What Windows version are you using?

@dalthviz, please check what's happening with our Windows installer locally.

**Comment by wCTcTU4P** [comment]
> Hey @wCTcTU4P, thanks for reporting. We'll take a look at this as soon as we can. What Windows version are you using?
> 
> @dalthviz, please check what's happening with our Windows installer locally.

I have Windows 10 Pro, think it is 21H2, will get exact build when I'm back at home later today or early tomorrow. I did launch the previous version of spyder with no issues just prior to downloading and installing 5.2 (from the startup update prompt).

Also was in a rush earlier, title was not well written, shouldn't say "installer". Or should say installer version. I think it was clear enough in the body, but the installation went fine, and it is actually launching spyder 5.2 from the windows installer version that is the issue.

**Comment by ccordoba12** [comment]
Ok, I reworded the title a bit, but feel free to change it if you think it doesn't reflect your problem correctly.

**Comment by yiorgosn** [comment]
Same problem here.!  Spyder ver 5.2.0 (used Win standalone installer) shuts down immediately after launch.  I was able to freeze the Spyder launch cycle and the problem seems to be that it can not find the spyder-kernels package with the new installation.

I uninstalled Spyder version 5.1.5 (as per your instructions on the website) and installed 5.2.0, Spyder launches for a brief second can NOT find the spyder-kernels module at the default location and shuts down.  I am using anaconda to manage the environment but I launch Spyder as standalone.  Spyder-kernels are present/installed in the (base) environment according to Anaconda so Spyder ver 5.2.0 is looking for them somewhere else.

Win 10 Pro  (21H2)
Python ver 3.9

**Comment by rhkarls** [comment]
If its of any help I'm running 5.2.0 from installer without this issue on Windows 10 Pro 1909, spyder-kernels 2.2.0 in separate environment. I had installed spyder-kernels 2.2.0 prior to updating Spyder itself.

However, if I change the settings in Spyder to default environment (same as Spyders install I get) I get an error message flashed (see gif) when starting a new kernel as well as:

```
 It seems the debugger cannot resolve D:\obj\windows-release\37amd64_Release\msi_python\zip_amd64\ntpath.py
 This may make the debugger miss breakpoints in the standard library.
 Related bug: https://bugs.python.org/issue1180193
```

![spyder_error_default_interpreter](https://user-images.githubusercontent.com/2330659/143562158-2a58cb8f-310c-465c-b1a7-46971dc93149.gif)

Next, if I set the environment to a custom one that does not have updated spyder-kernels to 2.2.0 I get the same error as others describe here when restarting Spyder (crashes on launch with issue reporter window briefly visible). Before restarting Spyder I tried opening a new IPython console with the environment having an older spyder-kernels and got the following traceback

```
Traceback (most recent call last):
  File "C:\Users\USERNAME\AppData\Local\Programs\Spyder\pkgs\spyder\plugins\ipythonconsole\widgets\main_widget.py", line 1746, in close_client
    client.shutdown(last_client)
  File "C:\Users\USERNAME\AppData\Local\Programs\Spyder\pkgs\spyder\plugins\ipythonconsole\widgets\client.py", line 585, in shutdown
    self.shellwidget.shutdown(shutdown_kernel)
  File "C:\Users\USERNAME\AppData\Local\Programs\Spyder\pkgs\spyder\plugins\ipythonconsole\widgets\shell.py", line 176, in shutdown
    self.kernel_manager.stop_restarter()
AttributeError: 'NoneType' object has no attribute 'stop_restarter'
```

**Comment by yiorgosn** [comment]
For clarification, I normally have Spyder set to use a custom environment namely, Anaconda's base (Python 3.9.7) and conda will not let me install spyder-kernels=2.0.0, my Anaconda base environment is using spyder-kernels=2.1.3.

**Comment by yiorgosn** [comment]
SOLUTION found for me.  I have Spyder using the Anaconda(base) environment and Spyder ver 5.2.0 wasn't able to find spyder-kernels ver 2.2.0 upon launch and would shut down immediately.  

The conda channels do not have spyder-kernels ver 2.2.0 yet.

I did a pip install spyder-kernels==2.2.0 (so now the package is not visible in anaconda) but Spyder ver 5.2.0 runs fine.

**Comment by wCTcTU4P** [comment]
> Ok, I reworded the title a bit, but feel free to change it if you think it doesn't reflect your problem correctly.

No worries! I was on my phone and hadn't noticed that the title was updated. If that was the wording that works best on your end, that's fine with me.

I just tried installing spyder-kernels 2.2.0 as yiorgosn mentioned, and it worked for me as well. That is in line with what I originally mentioned - seemingly as soon as the console initialization began previously, it crashed the GUI. Are there any foreseeable issues with sticking with spyder-kernels 2.2.0?
